File: packages/witness/src/templates/puppeteer/visual-example.test.js
// TestivAI Visual Regression Example
// Docs: https://testiv.ai/docs#puppeteer
const puppeteer = require('puppeteer');
const { witness } = require('../testivai-witness');

describe('Visual Regression', () => {
  let browser, page;

  beforeAll(async () => {
    browser = await puppeteer.launch({ args: ['--remote-debugging-port=9222'] });
    page = await browser.newPage();
  });

  afterAll(async () => await browser.close());

  it('captures homepage', async () => {
    await page.goto('http://localhost:3000');
    await witness(page, 'homepage');
  });
});

File: packages/witness/src/templates/selenium-testng/VisualExampleTest.java
// TestivAI Visual Regression Example
// Docs: https://testiv.ai/docs#java
package testivai;

import org.testng.annotations.*;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;

public class VisualExampleTest {
    WebDriver driver;

    @BeforeClass
    public void setup() {
        ChromeOptions options = new ChromeOptions();
        options.addArguments("--remote-debugging-port=9222");
        driver = new ChromeDriver(options);
    }

    @AfterClass
    public void teardown() {
        driver.quit();
    }

    @Test
    public void homepageLooksCorrect() {
        driver.get("http://localhost:3000");
        TestivAIWitness.witness(driver, "homepage");
    }
}
